lib/wordcounter.rb

```diff
@@ -0,0 +1,93 @@
+require 'delegate'
+
+class WordHistogram < SimpleDelegator
```

Comment:
Cool!  I haven't seen this used before in ruby.  Our python applications make heavy use of the decorator pattern, which is sort of similar.